lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1488,8 +1488,28 @@ DIFile *SPIRVToLLVMDbgTran::getFile(const SPIRVId SourceId) {
                      ParseChecksum(ChecksumStr));
   }
 
-  return getDIFile(getString(SourceArgs[FileIdx]), std::nullopt,
-                   getStringContinued(SourceArgs[TextIdx], Source));
+  std::optional<DIFile::ChecksumInfo<StringRef>> CS;
+  SPIRVWord StrIdx = SourceArgs[TextIdx];
+  if (Source->getExtSetKind() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200) {
+    if (!getDbgInst<SPIRVDebug::DebugInfoNone>(SourceArgs[ChecksumKind]) &&
+        !getDbgInst<SPIRVDebug::DebugInfoNone>(SourceArgs[ChecksumValue])) {
+      llvm::DIFile::ChecksumKind Kind = SPIRV::DbgChecksumKindMap::rmap(
+          static_cast<SPIRVDebug::FileChecksumKind>(
+              BM->get<SPIRVConstant>(SourceArgs[ChecksumKind])
+                  ->getZExtIntValue()));
+      StringRef Checksum = getString(SourceArgs[ChecksumValue]);
+      size_t ChecksumEndPos = Checksum.find_if_not(llvm::isHexDigit);
+      CS.emplace(Kind, Checksum.substr(0, ChecksumEndPos));
+    }
+
+    if (SourceArgs.size() == MaxOperandCount)
+      StrIdx = SourceArgs[TextNonSemIdx];
+    else
+      StrIdx = SPIRVID_INVALID;
```

Comment:
Mmm, I'm starting to like putting debug info none(s) less. If we have another SPIR-V generator it would mean, that reader will end up not processing this instruction correctly. The spec defines number of operands for each optional parameters combination. If we have 1 optional parameter - it's always just Source. If 2 - it's always Kind and Checksum. If 3 - it's Kind followed by Checksum followed by Source.

Reply:
Otherwise we should explicitly say in the spec, that Kind and Checksum are not optional, but mandatory and they can be DebugNoneId

Reply:
Ok, maybe you're right and it would be better. I just understood the "optional" meaning differently - that they could be not set, but should if there is a text operand afterwards.